tools/scran-normalize/scran-normalize.R

```diff
@@ -0,0 +1,64 @@
+# load packages that are provided in the conda env
+options( show.error.messages=F,
+       error = function () { cat( geterrmessage(), file=stderr() ); q( "no", 1, F ) } )
+loc <- Sys.setlocale("LC_MESSAGES", "en_US.UTF-8")
+warnings()
+
+library(optparse)
+library(scran)
+print("ok")
+
+#Arguments
+option_list = list(
+  make_option(
+    c("-d", "--data"),
+    default = NA,
+    type = 'character',
+    help = "Input file that contains count values to transform"
+  ),
+  make_option(
+    c("-s", "--sep"),
+    default = '\t',
+    type = 'character',
+    help = "File separator [default : '%default' ]"
+  ),
+  make_option(
+    c("-o", "--out"),
+    default = "res.tab",
+    type = 'character',
+    help = "Output name [default : '%default' ]"
+  )
+)
+
+opt = parse_args(OptionParser(option_list = option_list),
+                 args = commandArgs(trailingOnly = TRUE))
+
+if (opt$sep == "tab") {opt$sep = "\t"}
+print(opt$sep)
+data = read.table(
+  opt$data,
+  check.names = FALSE,
+  header = TRUE,
+  row.names = 1,
+  sep = opt$sep
+)
+
+## Import data as a SingleCellExperiment object
+sce <- SingleCellExperiment(list(counts=as.matrix(data)))
+
+## Compute sum factors
+sce <- computeSumFactors(sce)
+
+sce <- normalize(sce)
+
+logcounts <- data.frame(genes = rownames(sce), logcounts(sce), check.names = F)
```

Comment:
at final, what is the unit for the results of scran normalization ? cpm ? tpm ? something else I guess ?

Reply:
@bellenger-l any follow up on this ?

Reply:
@drosofff definitively not TPM because it doesn't take into account the gene length.
It creates a reference : so either the average of all transcriptomes (no clustering step) or if we choose a clustering step the average of each cluster. Then it pools cells and then sum their expression profiles. The size factor is described as the median ration between the count sums and the average across all genes. Finally it constructs a linear distribution (deconvolution method, Lun et al., 2016) of size factors by taking multiple pools of cells.

Reply:
OK, I see. So we could say that these are "scran normalized count values". Can you say me again whether the tool log-transforms the data (automatically or optionally). This is important to clearly states what it is for downstream analyses

Reply:
The normalize function from the scater package return by default the log transformed values.

Reply:
But we can choose to not log transformed the values

Reply:
good, but explain this not only to me. In the help section please. This is an essential piece of information to understand and to avoid huge bugs downstream of this tools